### PR TITLE
internal/v4: add cache-control headers for archive GET

### PR DESCRIPTION
**Pull request by rogpeppe**
This will allow the client to know that it can cache the archive files.

### CODE REVIEW

**Comment by frankban** [comment]
:+1: thank you!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/515/

File: internal/v4/archive.go [hunk 1]
```diff
@@ -78,6 +83,11 @@ func (h *Handler) serveArchive(id *charm.Reference, w http.ResponseWriter, req *
 	return nil
 }
 
+func setCacheControl(h http.Header, age time.Duration) {
+	seconds := int(age / time.Second)
+	h.Set("Cache-Control", "public, max-age="+strconv.Itoa(seconds))
```
Comment (mhilton) [inline internal/v4/archive.go:93]: might be worth adding some additional  headers, stale-while-revalidate would seem like a good candidate (http://tools.ietf.org/html/rfc5861).

**Comment by mhilton** [comment]
:+1: LGTM

File: internal/v4/archive.go (hunk 1 above)
Comment (rogpeppe) [inline internal/v4/archive.go:93]: agreed, but let's keep it simple for now.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/516/

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**